packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -81,7 +83,148 @@ class AnimatedIcon extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // TODO(amirh): implement this.
-    return new Container();
+    // TODO(amirh): implement semantics, text direction, scaling.
+    final _AnimatedIconData iconData = icon;
+    return new CustomPaint(
+      size: iconData.size,
+      painter: new _AnimatedIconPainter(
+        iconData.paths,
+        progress.value,
+        color,
+        () => new ui.Path(),
+      ),
+    );
   }
 }
+
+typedef ui.Path _UiPathFactory();
+
+class _AnimatedIconPainter extends CustomPainter {
```

Comment:
there's a more efficient way to build this where the CustomPainter actually gets the Listenable instead of using an AnimatedWidget and creating a new CustomPainter each time. If you search for "implements CustomPainter" you should find examples of doing this (i think the scrollbar or overscroll glow mechanisms use it).

Reply:
Thanks, done